src/main/java/ar/com/carloscurotto/storm/complex/topology/propagator/gloss/MessageSender.java

```diff
@@ -0,0 +1,118 @@
+
+    private Producer<String> messageProducer;
+
+    /**
+     * Constructs the MessageSender with the give message producer.
+     * 
+     * @param theMessageProducer
+     *            a {@link Producer<String>} that will be used to send the update messages to the consumer of the
+     *            producer.
+     */
+    public MessageSender(final Producer<String> theMessageProducer) {
+        Validate.notNull(theMessageProducer, "The message producer cannot be null");
+        messageProducer = theMessageProducer;
+    }
+
+    /**
+     * Creates the marshallers for the trade status messages classes and stores them in the marshallers map.
+     */
+    @Override
+    protected void doOpen() {
+        try {
+            marshallers.put(NormalTradeStatusMessage.class,
+                    JAXBContext.newInstance(NormalTradeStatusMessage.class).createMarshaller());
+            marshallers.put(ExcpTradeStatusMessage.class,
+                    JAXBContext.newInstance(ExcpTradeStatusMessage.class).createMarshaller());
+            marshallers.put(TradeCommentsMessage.class,
+                    JAXBContext.newInstance(TradeCommentsMessage.class).createMarshaller());
+        } catch (JAXBException e) {
+            throw new RuntimeException(
+                    "MessageSender can't create the marshallers needed to convert the trade objects to xml",
+                    e);
+        }
+
+    }
+
+    @Override
+    protected void doClose() {
+        messageProducer.close();
+    }
+
+    /**
+     * Sends the messages in theMessages parameters using the internal producer.
+     * 
+     * @param theMessages
+     *            a {@link List<TradeMessage>} with the messages to be sent.
+     * @return null.
+     */
+    @Override
+    protected Void doExecute(List<TradeMessage> theMessages) {
+        for (TradeMessage message : theMessages) {
+            send(message);
+        }
+        return null;
+    }
+
+    /**
+     * Sends a message after marshalling it into an xml string.
+     *
+     * @param theMessage
+     *            a TradeMessage instance. It cannot be null.
+     */
+    private void send(TradeMessage theMessage) {
+        Validate.notNull(theMessage, "The message cannot be null");
+        messageProducer.send(marshal(theMessage));
+    }
+
+    /**
+     * Returns a string representing the message parameter.
+     *
+     * @param theMessage
+     *            the message to marshal to a string. It cannot be null.
+     * @return the marshalled string representation of the message parameter.
+     */
+    private String marshal(TradeMessage theMessage) {
+        Validate.notNull(theMessage, "message cannot be null");
+
+        StringWriter writer = new StringWriter();
+        StreamResult result = new StreamResult(writer);
+
+        Marshaller marshaller = marshallers.get(theMessage.getClass());
+
+        try {
+            marshaller.marshal(theMessage, result);
```

Comment:
Tenes que cerrar el Stream en el finally ya que JAXB no lo cierra despues de usarlo.
http://blog.bdoughan.com/2011/08/jaxb-and-java-io-files-streams-readers.html